### ISSUE TITLE
[=BUG=] Song is normally end at mid of song

### ISSUE THREAD

**Issue by cjh980402** [play-dl/play-dl#32]
#### Describe the bug

 - Node.js version: 16.6.2
 - OS & version: ubuntu 20.04

Thank you for your kind feedback and light-weight youtube module! So now I am using your module for my discord bot.
In v0.7.3, there's no critical bug or error like uncaught error. However, when I play the long song (duration is longer than 1h), song is normally end at mid of song.

#### Actual behavior
- First song
- url: https://www.youtube.com/watch?v=COAudMKjcd8
- duration: 1h 39m 10s
- end time: about 32m

- Second song
- url: https://www.youtube.com/watch?v=uL5qScnXgEc
- duration: 2h 20m 5s
- end time: about 1h 34m

#### Expected behavior
- First song
- url: https://www.youtube.com/watch?v=COAudMKjcd8
- duration: 1h 39m 10s
- end time: 1h 39m 10s

- Second song
- url: https://www.youtube.com/watch?v=uL5qScnXgEc
- duration: 2h 20m 5s
- end time: 2h 20m 5s
 

#### Code to reproduce

```js

```

#### ScreenShots

![image](https://user-images.githubusercontent.com/23624124/131608037-5b51a1c2-93b0-4671-8233-19a1c3537f61.png)

#### Checklist

- [x] I have read the documentation.
- [x] I have tried my code with the latest version of Node.js and Play-dl.

**Comment by killer069** [comment]
For you first [url](https://www.youtube.com/watch?v=COAudMKjcd8), I got to see this.

So AudioResource ended at 2078020 [ aka 34 Min 38 s ] not 32 mins. And emitted Error is false. So it means no error either.

If you force close a resource, you will see that audio Resource will give you error [ Premature Close ].

But in this case, no errors. So need to run few more tests to see where the issue lies. Is it play-dl issue or some other issue.

```js
AudioResource {
  playbackDuration: 2078020,
  started: true,
  silenceRemaining: 0,
  edges: [
    {
      type: 'webm/opus demuxer',
      to: [Node],
      cost: 1,
      transformer: [Function: transformer],
      from: [Node]
    }
  ],
  playStream: WebmDemuxer {
    _readableState: ReadableState {
      objectMode: true,
      highWaterMark: 16,
      buffer: BufferList { head: null, tail: null, length: 0 },
      length: 0,
      pipes: [],
      flowing: null,
      ended: false,
      endEmitted: false,
      reading: true,
      constructed: true,
      sync: false,
      needReadable: true,
      emittedReadable: false,
      readableListening: false,
      resumeScheduled: false,
      errorEmitted: false,
      emitClose: true,
      autoDestroy: true,
      destroyed: true,
      errored: null,
      closed: true,
      closeEmitted: true,
      defaultEncoding: 'utf8',
      awaitDrainWriters: null,
      multiAwaitDrain: false,
      readingMore: false,
      decoder: null,
      encoding: null,
      [Symbol(kPaused)]: null
    },
    _events: [Object: null prototype] {
      prefinish: [Function: prefinish],
      close: [Array],
      end: [Function: onend],
      finish: [Function: onfinish],
      error: [Array]
    },
    _eventsCount: 5,
    _maxListeners: undefined,
    _writableState: WritableState {
      objectMode: false,
      highWaterMark: 16384,
      finalCalled: false,
      needDrain: false,
      ending: false,
      ended: false,
      finished: false,
      destroyed: true,
      decodeStrings: true,
      defaultEncoding: 'utf8',
      length: 0,
      writing: false,
      corked: 0,
      sync: false,
      bufferProcessing: false,
      onwrite: [Function: bound onwrite],
      writecb: null,
      writelen: 0,
      afterWriteTickInfo: null,
      buffered: [],
      bufferedIndex: 0,
      allBuffers: true,
      allNoop: true,
      pendingcb: 0,
      constructed: true,
      prefinished: false,
      errorEmitted: false,
      emitClose: true,
      autoDestroy: true,
      errored: null,
      closed: true,
      closeEmitted: true,
      [Symbol(kOnFinished)]: []
    },
    allowHalfOpen: true,
    _remainder: null,
    _length: 34062080,
    _count: 34061800,
    _skipUntil: null,
    _track: { number: 1, type: 2 },
    _incompleteTrack: {},
    _ebmlFound: true,
    [Symbol(kCapture)]: false,
    [Symbol(kCallback)]: null
  },
  metadata: null,
  silencePaddingFrames: 5,
  audioPlayer: undefined
}
```

**Comment by cjh980402** [comment]
Oh, you're right. It was almost 34mins. Is this issue need more test case?

**Comment by killer069** [comment]
> Oh, you're right. It was almost 34mins. Is this issue need more test case?

This requires some more test since I am not able to find that how stream is getting ended only after 34 mins.

**Comment by killer069** [comment]
Please update to 0.7.5. Since I messed up somethings

**Comment by cjh980402** [comment]
Okay, thanks.
What is difference with timeout duration 300 and 280?

**Comment by killer069** [comment]
> Okay, thanks.
> What is difference with timeout duration 300 and 280?

Actually, I made some assumptions wrong. 

As I told earlier (maybe), I get 2nd data after 280 seconds and then after on every 300 seconds.

So as in this case, it ended at 38 seconds so just 2 seconds behind my assumption. 

So I changed loop after 2nd data to get data every after 280 seconds. This will ensure to get data before stream actually reach that point.
